src/blueprint/from_config.rs

```diff
@@ -223,6 +225,56 @@ fn to_type(name: &str, list: &Option<bool>, required: &Option<bool>, list_type_r
   }
 }
 
+fn validate_query(config: &Config) -> Valid<()> {
+  let query_type_name = config
+    .graphql
+    .schema
+    .query
+    .as_ref()
+    .validate_some("Query root is missing".to_owned())?;
+
+  let Some(query) = config.find_type(query_type_name) else {
+    return Valid::fail("Query type is not defined".to_owned()).trace(query_type_name);
+  };
+
+  query
+    .fields
+    .iter()
+    // validate only required fields
+    // TODO: somehow when field is required it has a value of Option<false> due to https://github.com/tailcallhq/tailcall/blob/49e39e12013cf837ce0cc78562c4207f4a9782ad/src/config/from_document.rs#L195
```

Comment:
this is related to https://github.com/tailcallhq/tailcall/issues/371

I didn't find when the value could be `Some(true)` in this case so this could be a little messy

Reply:
- We could implement `is_required()` on field which unwrap_or(false)
- You could possibly attempt #371 first? I still don't have a PR yet.

What do you want to do?

Reply:
I prefer to fix it once and for all by removing option wrappers as it already hit me couple times for the last couple days 😄 

If I can find some time before someone solves it, I would like to attempt.

Reply:
Perfect! Looking forward to your PR for #371 :)